digdag-core/src/main/java/io/digdag/core/agent/OperatorManager.java

```diff
@@ -324,27 +324,34 @@ protected TaskResult callExecutor(Path projectPath, String type, TaskRequest mer
         return operator.run();
     }
 
-    private static class OperatorSecretFilter
-            implements Predicate<String>
+    private static class AgentOperatorSecretFilter
+            implements DefaultSecretProvider.OperatorSecretFilter
```

Comment:
it is not possible because an Operator instance is not available until OperatorManager calls `OperatorFactory.newOperator(OperatorContext)` but a DefaultSecretProvider.OperatorSecretFilter instance is required by `OperatorFactory.newOperator(OperatorContext)` because OperatorContext contains a SecretProvider instance that contains OperatorSecretFilter.